en/deploy/README.md

```diff
@@ -130,159 +130,118 @@ Your code is now on GitHub. Go and check it out!  You'll find it's in fine compa
+$ pip3.6 install --user pythonanywhere
 ```
 
+That should print out some things like `Collecting pythonanywhere`, and eventually end with a line saying `Successfully installed (...) pythonanywhere- (...)`.
 
-### Creating a virtualenv on PythonAnywhere
-
-Just like you did on your own computer, you can create a virtualenv on PythonAnywhere. In the Bash console, type:
+Now we run the helper to automatically configure our app from GitHub. Type the following into the console on PythonAnywhere (don't forget to use your GitHub username in place of `<your-github-username>`):
 
 {% filename %}PythonAnywhere command-line{% endfilename %}
 ```
-$ cd my-first-blog
-
-$ virtualenv --python=python3.6 myvenv
-Running virtualenv with interpreter /usr/bin/python3.6
-[...]
-Installing setuptools, pip...done.
-
-$ source myvenv/bin/activate
-
-(myvenv) $  pip install django~=1.11.0
-Collecting django
-[...]
-Successfully installed django-1.11.3
+$ pa_autoconfigure_django.py https://github.com/<your-github-username>/my-first-blog.git
 ```
 
+As you watch that running, you'll be able to see what it's doing:
 
-> **Note** The `pip install` step can take a couple of minutes.  Patience, patience!  But if it takes more than five minutes, something is wrong.  Ask your coach.
+- Downloading your code from GitHub
+- Creating a virtualenv on PythonAnywhere, just like the one on your own PC
+- Updating your settings file with some deployment settings
+- Setting up a database on PythonAnywhere using the "migrate" command
+- Setting up your static files (we'll learn about these later)
+- And configuring PythonAnywhere to serve your web app via its API
 
-<!--TODO: think about using requirements.txt instead of pip install.-->
+On PythonAnywhere all those steps are automated, but they're the same steps you
+would have to go through with any other server provider.  The main thing to notice
+right now is that your database on PythonAnywhere is actually totally separate from
+your database on your own PC -- that means it can have different posts and admin accounts.
 
-### Creating the database on PythonAnywhere
+As a result, just as we did on your own computer, we repeat the step to
+initialize the admin account with `createsuperuser`.
 
-Here's another thing that's different between your own computer and the server: it uses a different database. So the user accounts and posts can be different on the server and on your computer.
+First, we activate the PythonAnywhere virtualenv, and then we `cd` into our
+code directory.  PythonAnywhere has named them both after your site's name:
 
-Just as we did on your own computer, we repeat the step to initialize the database on the server, with `migrate` and `createsuperuser`:
 
 {% filename %}PythonAnywhere command-line{% endfilename %}
 ```
-(mvenv) $ python manage.py migrate
-Operations to perform:
-[...]
-  Applying sessions.0001_initial... OK
-(mvenv) $ python manage.py createsuperuser
+$ workon $USER.pythonanywhere.com
```

Comment:
Is it OK to use the `workon` command (from virtualenvwrapper, isn't it?) without introducing it first? Though, to manually source the `activate` script, we would have to know where the magic script has placed the virtualenv it created ...

Maybe add a short explanation that this does the same as the `source myvenv/bin/activate` we did locally.

Reply:
I think at least a short explanation would be a good idea yes.  I'll add one.

(`workon`  is from virtualenvwrapper yes.  we quite like virtualenvwrapper at pythonanywhere, because it puts all the virtualenvs in a sensible place, and it encourages you to give them all sensible names -- in this case, we name them after the site they're working on.  i think one of the problems with "manual" virtualenv creation is that they tend to all end up named "myenv" or "venv" or "virtualenv", and then you can't actually tell at a glance what you're working on...)

Reply:
How about I look into whether we could drop virtualenvwrapper for the autoconfigure script, and just have a "normal" virtualenv folder, in the same folder as the source code folder?  maybe a `--virtualenv-name` command line arg, then we can set it to "myenv", and it will be exactly like on the users' own laptop?

Reply:
It makes me wonder: would it be possible to fully abstract the pythonanywhere setup behind a script? Including pulling. That way we don't need to explain it and skipping would be easy as well. It's a comment I heard a few times: deployment to pythonanywhere is nice, but doesn't add a lot of value in terms of learning. A lot of duplicated steps. It is nice that they can actually go there on their phones so the end result is certainly nice but in a time of PaaS and even more modern alternatives, should we really start with teaching about hosting? The basics of web development are hard enough as it is.

Reply:
> would it be possible to fully abstract the pythonanywhere setup behind a script? Including pulling.

The `pa_autoconfigure_django.py` tool introduced with this change does that, doesn't it? (Assuming that by "pulling" you mean pulling the git repo from GitHub to pythonanywhere.)

Reply:
What I meant by pulling was indeed from Github to pythonanywhere, but in the context that there was an update after the initial deployment.

Reply:
> How about I look into whether we could drop virtualenvwrapper for the autoconfigure script, and just have a "normal" virtualenv folder, in the same folder as the source code folder? maybe a `--virtualenv-name` command line arg, then we can set it to "myenv", and it will be exactly like on the users' own laptop?

If the aim is for this tooling to be useful beyond the DjangoGirls tutorial, I wouldn't drop `virtualenvwrapper`, as there are good reasons to use it (and the benefits it brings) in other contexts.

A virtualenv created by virtualenvwrapper can be used just like one created any other way (i.e. by sourcing `bin/activate` instead of using `workon`). So, the only thing we need to make it accessible to DjangoGirls participants, is a way to control its location and name.

When `pa_autoconfigure_django.py` invokes `mkvirtualenv` it will explicitly pass the name. The location can be controlled with `$WORKON_HOME`, which  `pa_autoconfigure_django.py` could set and export for invoking `mkvirtualenv`.

Reply:
I wasn't suggesting to drop virtualenv, but rather replace every single manual interaction with pythonanywhere with a command line utility.

Reply:
> I wasn't suggesting to drop virtualenv

I know. @hjwp [suggested that](https://github.com/DjangoGirls/tutorial/pull/1190#discussion_r163512609).

> but rather replace every single manual interaction with pythonanywhere with a command line utility

You mean a utility to be run from the participant's local command line instead of the pythonanywhere command line?

Reply:
Indeed - not sure if that's possible with the API but if you can do an initial deployment ...

Reply:
re: a command-line tool for "update my deployment now", I think the idea definitely has potential, it could abstract doing git pull -- manage.py migrate -- manage.py collecstatic -- reload.

For now, the actual changes to the later deploy chapters are pretty minor, there's just one chapter in which we have to tell them to run `collectstatic`, so maybe it would be OK to defer this "auto-update" tool for a future release?  perhaps once we have our new single `pythonanywhere` executable ;-)

Reply:
re: making a deploy script that you could run from your own computer, that's currently not possible, but definitely something we aspire to.

Reply:
finally, re: "vanilla" virtualenv vs virtualenvwrapper, the suggestion that we "hack" virtualenvwrapper by changing `$WORKON_HOME` in order to essentially force it to create something that looks like a vanilla virtualenv ends up removing all of the reasons to use virtualenvwrapper I think, so I'm -1 on that.   

but I think we could definitely consider the option of adding a `--no-virtualenvwrapper` argument to the script (or maybe, heck, just a `--djangogirls` arguement) in order to make the virtualenv setup on pythonanywhere look more like the virtualenv on the users' laptop.

yes the script is intended to be useful for people outside of djangogirls, but djangogirls is my first target market, and it's more important to me to make y'all happy first, and I can worry about other people later.

Reply:
I wonder if a nice way to sidestep this issue for now would be to have the `pa_autoconfigure_django.py` script automatically drop people into a shell with the virtualenv active and `cd`'d into the right directory?  that's something that would be useful for almost anyone, and I think I know how to do it...  Will take a look now.

Reply:
> finally, re: "vanilla" virtualenv vs virtualenvwrapper, the suggestion that we "hack" virtualenvwrapper by changing $WORKON_HOME in order to essentially force it to create something that looks like a vanilla virtualenv ends up removing all of the reasons to use virtualenvwrapper I think, so I'm -1 on that.

What I meant was to do that
1. only **temporarily** (for the invocation of virtualenvwrapper in `pa_autoconfigure_django.py`)<br />and
2. only when explicitly requested (i.e. when `pa_autoconfigure_django.py` is called with a CLI option towards that end)

Thus
```bash
pa_autoconfigure_django.py https://github.com/<your-github-username>/some-django-project.git
```
would have the same behavior as it currently has, and for the tutorial, we'd do something like

```bash
pa_autoconfigure_django.py https://github.com/<your-github-username>/my-first-blog.git --virtualenv-name myvenv --workon-home .
```
or even
```bash
WORKON_HOME="." pa_autoconfigure_django.py https://github.com/<your-github-username>/my-first-blog.git --virtualenv-name myvenv
```

> I wonder if a nice way to sidestep this issue for now would be to have the `pa_autoconfigure_django.py` script automatically drop people into a shell with the virtualenv active and `cd`'d into the right directory?

👍 Yeah, if that's workable, it's probably an even better approach.

✨ Hide the magic behind sufficiently advanced technology, so it won't unsettle anyone. ✨

Reply:
:) working on said magic now 🎩

Reply:
OK magic done!